build/Jenkinsfile.pr

```diff
@@ -0,0 +1,261 @@
+#!/usr/bin/env groovy
+
+@Library('shared-pipelines') _
+import org.mattermost.Utils
+import java.util.*;
+def utils = new org.mattermost.Utils()
+def rnd = UUID.randomUUID().toString()
+
+pipeline {
+    agent {
+        label 'default-mm-builder'
+    }
+
+    options { buildDiscarder(logRotator(numToKeepStr: '3')) }
+
+    environment {
+        DCNAME = "${rnd}"
+    }
+
+    stages {
+        stage('Stop Old Builds') {
+            steps {
+                script {
+                    utils.stopOldBuilds()
+                }
+            }
+        }
+
+        stage('Checkout'){
+            steps {
+                cleanWs notFailBuild: true
+                sh """
+                    mkdir -p src/github.com/mattermost/mattermost-server
+                    mkdir -p src/github.com/mattermost/mattermost-webapp
+                    mkdir -p src/github.com/mattermost/enterprise
+                """
+                dir('src/github.com/mattermost/mattermost-server') {
+                    checkout scm
+                }
+                dir('src/github.com/mattermost/mattermost-webapp') {
```

Comment:
It would be nice if the server PR build just used the latest webapp instead of building it. No need to complicate this now, just something to think about.

Reply:
it gets the latest webapp from master or from the same branch name. it will build only if the hash is not there yet. (happens when you open both pr at the same time or the webapp is currently building)